### DISCUSSION TITLE
How to properly handler http 1.1 Server Response Header - `keep-alive: timeout=5`?

### DISCUSSION THREAD

**Discussion by Slach** [curl/curl#14487] [Q&A] [answered]
NodeJs introduce `Keep-Alive: timeout=X` hint in https://github.com/nodejs/node/issues/34560

All `nodejs` projects which use standard lib `http` server send this header by default

I'm struggling with the following bug, 
when connect between `Azurite` which uses `nodejs` and ClickHouse which uses `Azure-sdk-cpp` which use `libcurl` 
then `libcurl` inside `Azure-sdk-cpp` doesn't handle server header `Keep-Alive: timeout=5` and trying to write to an already closed TCP connection and got RST and send unnecessary parasite traffic

Unfortunately, I'm not successful with 
https://github.com/Azure/Azurite/issues/2053, https://github.com/ClickHouse/ClickHouse/issues/60447 
https://github.com/Azure/azure-sdk-for-cpp/issues/5877

@bagder 
Could you suggest properly `libcurl` initialization code snippet which allow proper handle server header `Keep-Alive: timeout=5` and doesn't use already closed TCP connection from server side?

**Comment by dfandrich** [comment]
If you're just interested in detecting that header and parsing it in your code,
you should be able to do that by setting a CURLOPT_HEADERFUNCTION and parsing
it when it comes in. You can then take the parsed value and set
CURLOPT_MAXLIFETIME_CONN or CURLOPT_TCP_KEEPINTVL or whatever makes sense for
the meaning of that header on future easy handles.

**Comment by bagder** [comment, marked as answer]
`Keep-Alive: timeout=X` is not a standard HTTP response header. libcurl does not look for it and does not care about it.

In addition to the option @dfandrich offers, I think maybe [CURLOPT_MAXAGE_CONN](https://curl.se/libcurl/c/CURLOPT_MAXAGE_CONN.html) is closes to this timer, even if it then applies to all connections the libcurl handle manages.

**Comment by Slach** [comment]
yep ;( https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Keep-Alive specifications which refers for this header is not clean
